app/components/task_list_items/check_red_line_boundary_component.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module TaskListItems
+  class CheckRedLineBoundaryComponent < TaskListItems::BaseComponent
```

Comment:
I've tried to copy the logic in this component faithfully from the presenter - it's a bit hairy!